File: Tutoriels/PHP/Projets/php/class/GProcessPrint.php
<?php   
    class GProcessPrint extends GProcess {
        //===============================================
        private static $m_instance = null;
        //===============================================
        private function __construct() {
        
        }
        //===============================================
        public static function Instance() {
            if(is_null(self::$m_instance)) {
                self::$m_instance = new GProcessPrint();  
            }
            return self::$m_instance;
        }
        //===============================================
        public function run() {
			GPrint::Instance()->show("Bonjour tout le monde"); 
        }
        //===============================================
    }
?>

File: Tutoriels/Java/Projets/code/GProject/src/manager/GProcessPrint.java
//===============================================
package manager;
//===============================================
public class GProcessPrint extends GProcess {
	//===============================================
	private static GProcessPrint m_instance = null; 
	//===============================================
	public GProcessPrint() { 

	} 	
	//===============================================
	public static GProcessPrint Instance() { 
		if(m_instance == null) { 
			m_instance = new GProcessPrint(); 
		}
		return m_instance; 
	} 
	//===============================================
	public void run() {
        GPrint.Instance().print("Bonjour tout le monde\n");
	}
	//===============================================
}
//===============================================
